src/js/modules/users/selectors.js

```diff
@@ -1,46 +1,67 @@
 import { createSelector } from "reselect";
+import appHistory from "../../tools/appHistory";
 
 export const getUsers = state => state.users.users;
 export const getRoles = state => state.users.roles;
 export const getUserRoles = state => state.users.userRoles;
 
 const getRequestedContributorSlug = (state, props) => props.match.params.contributor_slug;
+const getRequestedIllustratorSlug = (state, props) => props.match.params.illustrator_slug;
+const getRequestedPhotographerSlug = (state, props) => props.match.params.photographer_slug;
+
 const getRoleFromProps = (state, props) => props.role;
-const getUsersResponse = state => state.users.response;
 
 export const getContributorFromSlug = createSelector(
-  [ getUsers, getRequestedContributorSlug ],
-  (users, requestedContributorSlug) => {
-    return users[ requestedContributorSlug ];
+  [ getUsers, getUserRoles, getRequestedContributorSlug ],
+  (users, userRoles, requestedContributorSlug) => {
+    const user = Object.values(users).find(user => {
+      return user.slug === requestedContributorSlug;
+    });
+    if (userRoles.find(userRole => userRole.userId === user.id && userRole.roleId === 0)) {
+      return user;
+    }
+    appHistory.push('/the-uncompleted-404-page');
```

Comment:
should I make an action that uses the reducer to redirect? wait but it's still selectors...so no dispatch? nichhhhh wherreee